instrumentation/http/src/main/java/brave/http/HttpAdapter.java

```diff
@@ -19,7 +19,7 @@
 /** @deprecated Since 5.10, use {@link HttpRequest} and {@link HttpResponse} */
 @Deprecated public abstract class HttpAdapter<Req, Resp> {
   /** @see HttpRequest#method() */
-  @Nullable public abstract String method(Req request);
+  public abstract String method(Req request);
 
   /** @see HttpRequest#path() */
   @Nullable public String path(Req request) {
```

Comment:
Agree that HTTP spec doesn't actually require a path (OPTIONS commonly has * which I don't know if we call a path, CONNECT has only authority). So nullable seems fine.

But not so sympathetic with the availability bug issue - seems like bad http library as opposed to bad instrumentation, not sure if that specifically should fall into our defensive requirements.

Reply:
This was one example of path missing, something from last week, not the only time.

I think it is wise to be cautious as instrumentation should do no harm. Users can't always control versions of things and for example, even in armeria there are availability checks to prevent a known class of bugs. I don't expect you to agree with me but I disagree that we should just trust path will exist and anything that causes it not to be should crash requests.

Reply:
I reworded the javadoc towards the method rationale without suggesting that's the only reason why. The other reasons could be in the class of parse failures, or availability drift outside our control, but IMHO shouldn't cause requests to crash. hope the words balance!

Reply:
concretely, if we "fixed path" now, we'd definitely blow up former versions of instrumentation, causing revlock.

Reply:
PS I'd be fine with forcing path to be present on the methods relevant in a major release. We'd just know this would invalidate third party stuff and they'd get revlocked to the old one, causing the old version to need patches in worst case. Not signing up for that work personally, but agree in a green field or known break scenario stricter rules could apply.